python/tests/pytest.json

```diff
@@ -0,0 +1,54 @@
+[
+    {
+        "data_func": "constant_function",
+        "data_func_args": [2000,5,1,100],
+        "assert_func": "assert_prediction",
+        "assert_func_args": {
+            "expected_value": 5,
+            "threshold":0.5
+        },
+        "grid":   {
+            "#base": ["-P 50000 --preserve_performance_counters --save_resume "],
+            "#reg": ["", "--coin"]
+        },
+        "output": ["--readable_model", "-p"]
+    },
+    {
+        "data_func": "constant_function",
+        "data_func_args": [2000,5,1,100],
+        "assert_func": "assert_weight",
+        "assert_func_args": {
+            "expected_weights":[5, 0],
+            "atol": 1,
+            "rtol": 0.01
+        },
+        "grid":   {
+            "#base": ["-P 50000 --preserve_performance_counters --save_resume "],
+            "#reg": ["", "--coin"]
+        },
+        "output": ["--readable_model", "-p"]
+    },
+    {
+        "data_func": "constant_function",
+        "data_func_args": [2000,5,1,100],
+        "assert_func": "assert_weight",
+        "assert_func_args": {
+            "expected_weights":[5, 0],
+            "atol": 100,
+            "rtol": 10
+        },
+        "grid":   {
+            "#base": ["-P 1 --preserve_performance_counters --save_resume"],
+            "#reg": ["", "--coin", "--ftrl", "--pistol"]
+        },
+        "output": ["--readable_model", "-p"],
+        "*" : {
```

Comment:
not sure if I follow the syntax here: seems like this is outside of "grid" object?

Reply:
so bascially * allow users to have multiple tests so for example, 
if  "--learning_rate": [0.01, 0.001, 0.1] is in *, the it will created a grid object with each of these values in --learning_rate.
if there are multiple parameters in *, then it will produce all combination for example:
 {parameters in grid}  --learning_rate 0.01 --loss_function absolute ..
 {parameters in grid}  --learning_rate 0.001 --loss_function absolute ..
 {parameters in grid}  --learning_rate 0.01 --loss_function quantile ..
 {parameters in grid}  --learning_rate 0.001 --loss_function quantile ..